src/test/java/org/jooq/lambda/SeqTest.java

```diff
@@ -1543,6 +1543,22 @@ public void testIterate() {
     }
 
     @Test
+    public void testIterateWhilePresent() {
+        assertEquals(
+              asList(),
+              Seq.iterateWhilePresent(null, i -> Optional.empty()).toList()
+        );
```

Comment:
An `Optional` is a collection of arity `0..1`, which is why it makes sense to wrap it in a `Seq`. Likewise, JDK 9 will support [`Optional.stream()`](http://download.java.net/java/jdk9/docs/api/java/util/Optional.html#stream--). I wish it even implemented `Iterable`.

In that sense, it might be consistent to allow it as a parameter, but why do that? Why even call `iterate` when we *know* the resulting `Seq` will be empty?I mean, if the `seed` were `Supplier<Optional<T>>`, it might make sense (I wouldn't like it either), but with a hard-wired `Optional`, I think it's better to just not call the method, and value consistency with the existing `iterate()` method higher